src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -42,19 +54,47 @@ internal abstract class AbstractSnippetExpansionClient : ForegroundThreadAffinit
         protected readonly ITextView TextView;
         protected readonly ITextBuffer SubjectBuffer;
 
+        private readonly ImmutableArray<Lazy<ArgumentProvider, OrderableLanguageMetadata>> _allArgumentProviders;
+        private ImmutableArray<ArgumentProvider> _argumentProviders;
+
         protected bool indentCaretOnCommit;
         protected int indentDepth;
         protected bool earlyEndExpansionHappened;
 
-        internal IVsExpansionSession ExpansionSession;
+        private State _state = new() { _arguments = ImmutableDictionary.Create<string, string>() };
```

Comment:
is all updating of this serialized?  or on hte UI thread?  can you doc?

Reply:
Will document that writes only occur on the main thread.